### ISSUE TITLE
feature request: ? should show help screen

### ISSUE THREAD

**Issue by lavie** [sodapopcan/vim-twiggy#1]
When in the Twiggy "browser", it would be very helpful to hit `?` and get some succinct shortcut list. Most similar plugins offer this.

**Comment by sodapopcan** [comment]
When choosing mappings for non-modifiable buffers, I try and only chose those that--no matter how innocuous--won't break the user's expectations of how any ol' non-modifiable buffer works.  The only plugin I use that has this feature is nerdtree and even though it's not something I do often, it drives me nuts when I hit `?` to jump to up to a node and get the quickhelp instead.

Furthermore, I don't feel there are enough mappings to really warrant adding more noise to the codebase when `:h twiggy-mappings` gives a good, concise list.  Being my first plugin, I went a little crazy adding features--I use less than half of them myself!  Twiggy really isn't meant to be a branching power tool and I'd prefer to encourage people to always read the help, figure out the few things they are going to use and forget about it.

Of course, if there is ever enough demand for this I'll consider adding it, so I'll leave this open.

Otherwise, thanks for using the plugin and commenting!  This is my very first OS-issue on one of my own projects, haha.

**Comment by yoyoyeti** [comment]
I understand your reasoning for not adding it, but second the feature at the same time. I think it would be nice to have.

**Comment by sodapopcan** [comment]
@lavie @zac-harwood All right, you've convinced me.

I've got a first pass at it in the [`quickhelp`](https://github.com/sodapopcan/vim-twiggy/tree/quickhelp) branch.  If you guys would be so kind as to try it out for a couple of days and give me some feedback, that would be much appreciated.

**Comment by yoyoyeti** [comment]
@sodapopcan Awesome! I'd be glad to do some testing. The only issue I've noticed so far is that it seems to create a new '[No Name]' buffer every time I open/close the help menu. I'll let you know if I find anything else though.

**Comment by sodapopcan** [comment]
Right you are!  I just pushed a fix.  I wrote it in semi-haste.  I'll play with it some more tomorrow.  Thanks a lot for your help!

**Comment by sodapopcan** [comment]
One question I have is if `q` in quickhelp should close out the whole split or take you back to the branch buffer (ie, alias for `?`).  My original thought was to take you back to the branch buffer but as I was developing, I was wanting to close it quickly so now my opinion is a little skewed.  What do you think?

**Comment by yoyoyeti** [comment]
I think it makes sense that it would close the whole split.

**Comment by sodapopcan** [comment]
Cool cool, that's how I've left it.  I've been playing with it a bit today and it doesn't seem to be causing any issues.  Think I'm going to squash it onto master.

**Comment by sodapopcan** [comment]
Thanks for your help!

**Comment by sodapopcan** [comment]
@zac-harwood I've added a "press ? for help" message to master (86b87044eb59a1ed66dac2da3999d2edc6ca0e5d) that can be configured off, as well as a recommendation to read `:help twiggy-mappings` in the quickhelp buffer (which can also be configured off)  (df1e9d88f1b28d54488accf85799f1ccbb8b9796).  It's on master if you want to check it out.

**Comment by yoyoyeti** [comment]
@sodapopcan Awesome! I love it <3 The only other thing that I noticed (and this is super picky so it's not a big deal) is that if you press 'q' from the quickhelp menu it will close the quickhelp buffer but not the branches buffer. Other than that it seems golden.

**Comment by sodapopcan** [comment]
Oh shit, what?  Is it not replacing the branches buffer for you when you open it??  Like is it opening beside it or something?

**Comment by yoyoyeti** [comment]
Yeah, when I open the quickhelp menu it's still keeping the branches buffer open.

**Comment by sodapopcan** [comment]
Oh, well that's not supposed to happen.  Can you share your set up?  I took a quick look through your dotfiles repo (and was wholly unaware or merginal 😳) to see if there was anything obvious that might be causing it.  Do you mind sharing other aspects of your set up (OS and vim version mostly)?  I tried in console and macvim both of them are rendered in the branches buffer for me.

**Comment by sodapopcan** [comment]
Of course, it could be my setup too.  I often forgo plugin-dev 101 and don't test in a bare-bones set up.  I'll take look later on today.

**Comment by yoyoyeti** [comment]
I've got NVIM v0.2.2 and I'm running macOS High Sierra v10.13.3. I also went ahead and got my dotfiles repo up to date with my current setup although I don't think anything significant changed. It could totally just be my setup.

**Comment by sodapopcan** [comment]
I guess I should probably open a new issue for this as opposed to commenting on a closed one, but I tried in vim 8 and nvim with a vimrc that had only plug, fugitive, and twiggy and this didn't happen for me.  I then tried using your vimrc but things didn't immediately work in general (I have nvim installed but never use it) so I gave up for now as I have other stuff I have to do tonight.  It does appear to be a problem (or feature? haha) with your setup, though.

I'm still mega curious as to what it is and will probably look back into it later.  Lemme know if you find anything!
